crates/api/src/sitemap.rs

```diff
@@ -0,0 +1,50 @@
+use actix_web::{
+  http::header::{self, CacheDirective},
+  web::Data,
+  HttpResponse,
+};
+use chrono::{DateTime, FixedOffset};
+use lemmy_api_common::context::LemmyContext;
+use lemmy_db_schema::{source::post::Post, utils::DbPool};
+use lemmy_utils::error::LemmyResult;
+use sitemap_rs::{url::Url, url_set::UrlSet};
+use tracing::info;
+
+async fn generate_urlset(pool: &mut DbPool<'_>, limit: i64) -> LemmyResult<UrlSet> {
+  info!("Generating sitemap with latest {} posts...", limit);
+
+  let posts = Post::list_for_sitemap(pool, limit).await?;
+
+  info!("Loaded latest {} posts", posts.len());
+
+  let mut urls = vec![];
+  for post in posts {
```

Comment:
I'm fairly certain this could be done with `.map` and `collect` above, rather than this loop. But performance would probably be the same.

Reply:
Good idea, this is much cleaner now using `map_while`.